test/CMakeLists.txt

```diff
@@ -110,11 +110,37 @@ set_tests_properties(oqs_endecode
 )
 endif()
 
+add_executable(oqs_test_multi_thread oqs_test_multi_thread.c)
+target_link_libraries(oqs_test_multi_thread PRIVATE ${OPENSSL_CRYPTO_LIBRARY} ${OQS_ADDL_SOCKET_LIBS})
+if (NOT MSVC)
+    find_package(Threads REQUIRED)
+    target_link_libraries(oqs_test_multi_thread PRIVATE Threads::Threads)
+endif()
+
+add_test(
+  NAME oqs_multi_thread
+  COMMAND oqs_test_multi_thread
+          "oqsprovider"
+          "${CMAKE_CURRENT_SOURCE_DIR}/openssl-ca.cnf"
+)
+# openssl under MSVC seems to have a bug registering NIDs:
+# It only works when setting OPENSSL_CONF, not when loading the same cnf file:
```

Comment:
This comment to me warrants the question whether raising an issue in OpenSSL wouldn't be sensible?